apps/backend/src/modules/project-api-keys/project-api-key.service.ts

```diff
@@ -0,0 +1,123 @@
+import { ExceptionMessage } from "~/libs/enums/enums.js";
+import { type Encryption } from "~/libs/modules/encryption/encryption.js";
+import { HTTPCode } from "~/libs/modules/http/http.js";
+import { type Token } from "~/libs/modules/token/token.js";
+import { type Service } from "~/libs/types/types.js";
+
+import { ProjectApiKeyError } from "./libs/exceptions/exceptions.js";
+import {
+	type ProjectApiKeyCreateRequestDto,
+	type ProjectApiKeyCreateResponseDto,
+} from "./libs/types/types.js";
+import { ProjectApiKeyEntity } from "./project-api-key.entity.js";
+import { type ProjectApiKeyRepository } from "./project-api-key.repository.js";
+
+type Constructor = {
+	encryption: Encryption;
+	projectApiKeyRepository: ProjectApiKeyRepository;
+	token: Token;
+};
+
+class ProjectApiKeyService implements Service {
+	private encryption: Encryption;
+
+	private projectApiKeyRepository: ProjectApiKeyRepository;
+
+	private token: Token;
+
+	public constructor({
+		encryption,
+		projectApiKeyRepository,
+		token,
+	}: Constructor) {
+		this.projectApiKeyRepository = projectApiKeyRepository;
+		this.token = token;
+		this.encryption = encryption;
+	}
+
+	public async create(
+		payload: { userId: number } & ProjectApiKeyCreateRequestDto,
+	): Promise<ProjectApiKeyCreateResponseDto> {
+		const { projectId, userId } = payload;
+
+		const isProjectApiKeyExist = Boolean(
+			await this.projectApiKeyRepository.findByProjectId(projectId),
+		);
```

Comment:
let's do this as in other places
```suggestion
		const existingProjectApiKey = await this.projectApiKeyRepository.findByProjectId(projectId);
```